### ISSUE TITLE
Add Live Activities for Download % and/or speed (MB/s)

### ISSUE THREAD

**Issue by plyght** [kiwix/kiwix-apple#1037]
Exactly what the title says; Add live activities support for iOS to display download progress (speed and percentage completed) on the lock screen and in the Dynamic Island. This would provide real-time updates without needing to open the app.

**Comment by kelson42** [comment]
@rgaudin @BPerlakiH Clear for you what is requested?

**Comment by rgaudin** [comment]
It's a sort of system _notification_ that stays active for _live activities_ so the user can peak at progress without re-entering the app. It's common for sports app for instance.

**Comment by BPerlakiH** [comment]
Yes it is clear for me, we are talking about this Apple feature: https://developer.apple.com/design/human-interface-guidelines/live-activities

**Comment by kelson42** [comment]
The idea of implement Activities for downloads is validated

**Comment by plyght** [comment]
@kelson42 @BPerlakiH Any updates on this?

**Comment by kelson42** [comment]
@plyght The idea is validated, you want to implement it?

**Comment by plyght** [comment]
@kelson42 Do you mean coding the implementation myself or overseeing the feature’s development?

**Comment by kelson42** [comment]
@plyght Yes, I mean you making a PR to implement the feature.

**Comment by plyght** [comment]
@kelson42 see https://github.com/kiwix/kiwix-apple/pull/1058

**Comment by BPerlakiH** [comment]
@kelson42 

## UI

We have a couple of UI components to implement, here's what Safari is using (more or less):

<img width="397" alt="Image" src="https://github.com/user-attachments/assets/b21637f2-312e-418d-b2e8-ed98db16e62a" />
<img width="854" alt="Image" src="https://github.com/user-attachments/assets/34014ace-2e08-4bb9-8d1b-10240c7b36dd" />
<img width="1373" alt="Image" src="https://github.com/user-attachments/assets/22ce5551-4c82-47ea-b722-141f27a6339a" />

There's also a figma design template for these activities:
https://www.figma.com/community/file/1367915437752334285


## Support

At the moment our min. iOS version is 16.0. In order to support live activity, we need to change it to 16.6 (at least). This part should be OK.
If we want to add a similar button to pause / stop the download, [that needs at least iOS 17.0](https://www.figma.com/community/file/1367915437752334285).
That can be added conditionally as well. So below iOS 17.0, there won't be any button.

**Comment by kelson42** [comment]
@BPerlakiH LGTM
